components/base/wheeled/wheeled_base.go

```diff
@@ -91,17 +100,182 @@ type wheeledBase struct {
 	wheelCircumferenceMm int
 	spinSlipFactor       float64
 
-	left      []motor.Motor
-	right     []motor.Motor
-	allMotors []motor.Motor
+	left                []motor.Motor
+	right               []motor.Motor
+	allMotors           []motor.Motor
+	movementSensor      []movementsensor.MovementSensor
+	orienationSupported bool
+	orientationSensor   movementsensor.MovementSensor
+
+	opMgr  operation.SingleOperationManager
+	logger golog.Logger
+	model  referenceframe.Model
 
-	opMgr operation.SingleOperationManager
+	activeBackgroundWorkers sync.WaitGroup
 }
 
-func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
-	ctx, done := base.opMgr.New(ctx)
-	defer done()
+// createWheeledBase returns a new wheeled base defined by the given config.
+func createWheeledBase(
+	ctx context.Context,
+	deps registry.Dependencies,
+	cfg *Config,
+	baseName string,
+	logger golog.Logger,
+) (base.LocalBase, error) {
+	base := &wheeledBase{
+		widthMm:              cfg.WidthMM,
+		wheelCircumferenceMm: cfg.WheelCircumferenceMM,
+		spinSlipFactor:       cfg.SpinSlipFactor,
+		logger:               logger,
+	}
 
+	if base.spinSlipFactor == 0 {
+		base.spinSlipFactor = 1
+	}
+
+	for _, name := range cfg.Left {
+		m, err := motor.FromDependencies(deps, name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "no left motor named (%s)", name)
+		}
+		positionReporting, err := m.Properties(ctx, nil)
+		if positionReporting["PositonReporting"] && err == nil {
+			base.logger.Debugf("have encoders on motor %#v", m)
+		}
+		base.logger.Debug("no encoders found on left motors")
+		base.left = append(base.left, m)
+	}
+
+	for _, name := range cfg.Right {
+		m, err := motor.FromDependencies(deps, name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "no right motor named (%s)", name)
+		}
+		positionReporting, err := m.Properties(ctx, nil)
+		if positionReporting["PositonReporting"] && err == nil {
+			base.logger.Debugf("have encoders on motor %#v", m)
+		}
+		base.logger.Debug("no encoders found on right motors")
+		base.right = append(base.right, m)
+	}
+
+	for _, msName := range cfg.MovementSensor {
```

Comment:
why does this config accept a list of movement sensor names? It appears that it just sets the `orientationSensor` to the last one and only the first element of `movementSensor` is ever used (and only when `orientationSensor` exists, but the base's `orientationSensor` is never used)?

Reply:
future-proofing for when we might have `"movement_sensor": ["imu-wit", "gps-rtk", "rpLidar", "merged"]`.
You can use a mix of movement sensors for position, orientation, velocity etc. This is simply part 1 of many PRs to come. Next is `angularVelocity` for fixing the SetVelocity function, which could be from one or many gyros or imus.

It makes sense to take a list. This won't change. I should change `base.OrientationSensor[0]` however. Good catch.

Reply:
ok, understood. Can disregard